### PR TITLE
RNA: Improve `Button` loading layout

### PR DESCRIPTION
**Pull request by retrofox**
Improves `Button` state loading styles.

Part of https://github.com/Automattic/jetpack/issues/23708

#### Changes proposed in this Pull Request:

* Update loading layout for `Button`.

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No.

### `<Button />`
* Use Storybook
* Go to Button story
* Confirm you see a new default story there
* Check how the component behaves when changing its props values via the Controls panel

https://user-images.githubusercontent.com/77539/163185361-e9fdd384-79ba-4bd0-8702-f1e93e7d9a05.mov

### CODE REVIEW

**Comment by renatoagds** [comment]
We modified the initial layout proposal for Loading, based on this conversation: p1649858394876719-slack-C02M27K4741

TLDR; It now hides the `text` instead of keeping it together with `spinner`.

**Review by renatoagds — APPROVED** [review]
LGTM

**Comment by retrofox** [comment]
> Great news! One last step: head over to your WordPress.com diff, D78690-code, and deploy it.

r243665-wpcom
